File: trinta/src/app/driver-rider/Driver.ts
export interface Driver {
  id: string;
  nom: string;        // Changé de 'name' à 'nom'
  prenom: string;     // Ajouté
  vehicleType: string;
  capacity: number;
  latitude: number;
  longitude: number;
  rating: number;
  pricePerKm: number;
  experienceYears: number;
  responseTime: number;
  available: boolean;
  numTel: string;     // Changé de 'phone' à 'numTel'
  imageUrl: string;   // Changé de 'profilePicture' à 'imageUrl'
}

File: backendIA-ons/src/main/java/org/example/backendia/entities/Driver.java
package org.example.backendia.entities;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "chauffeurs")
@PrimaryKeyJoinColumn(name = "user_id")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Driver extends  User {
    private Double latitude;
    private Double longitude;
    private String vehicleType;
    private Integer capacity;
    private Double rating;
    private Double pricePerKm;
    private Integer experienceYears;
    private Integer responseTime;
    private Boolean available;

}
